docs/configuration/configuration.md

```diff
@@ -850,6 +850,16 @@ tls_config:
 namespaces:
   names:
     [ - <string> ]
+
+# Optional label and field selectors to limit the discovery process to a subset of available resources. 
+# See https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/
+# and https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ to learn more about the possible 
+# filters that can be used. Endpoints role supports pod, service and endpoints selectors, other roles
+# only support selectors matching the role itself (e.g. node role can only contain node selectors).
```

Comment:
I think this needs to document more the trade-off people are making here. Defining a label or field selector prevents Prometheus from re-using a single list/watch for all scrape configs (Prometheus already does this optimization today). Meaning that if there are two scrape configs each doing Prometheus discovery, then in reality Prometheus only opens 1 list/watch and processes the items through both relabeling configs, when this is used, the trade-off we're doing is we take the load off of Prometheus and put it onto the Kubernetes API, and create a list/watch per selector combination. It depends on the individual situation what may be the "best" approach. So I think this should have more thorough documentation about this trade-off.

Reply:
That's the case for other discovery mechanisms too, so a more general note up top could do it.

Reply:
It has been addressed inside this part only. I am open to move that as a general note but that would deserve a dedicated discussion as now the documentations is basically 'flat', tls_config is at the same level as `kube_sd_config`, so that would probably take some refactoring.